content/guides/virtualizing/bhyve.md

```diff
@@ -0,0 +1,158 @@
+tags = []
++++
+
+Virtual instances of operating systems are perfect for all kinds of testing purposes that need to be done in a safe and isolated environment. Installing Haiku in a virtual machine is a solution for people who do not want to install it on their physical computers, but wish to become familiar with it.
+
+In this guide the Haiku operating system is being run under virtual circumstances using FreeBSD and Bhyve.
+
+In this guide, we will be using an Anyboot image - it can be obtained [here](/get-haiku).  Both the ISO and anyboot images are available there, do select the closest mirror to enjoy higher transfer rates. Verify using the checksums to make sure that the downloaded files are not corrupted as they are big files.
+
+This guide assumes that you have installed FreeBSD on your device, you have made some basic configurations to your system and you are currently logged in as a user with root access. The Haiku virtual machine will be installed in UEFI mode.
+
+##### Go to section
+
+* [Setting up FreeBSD & Bhyve](#part_bhyve)
+* [Downloading Haiku](#part_download)
+* [Installing and running Haiku from an Anyboot image](#part_iso)
+
+### Preparing FreeBSD & Bhyve <a name="part_bhyve"></a>
+
+*Considering that FreeBSD is an operating system that is primarily used in server-based applications that do not rely on a graphical environment, this section was added for beginners that are not used to Unix-like terminals. This process is not much different in FreeBSD than in Linux distributions, so feel free to skip this section if you believe that you are capable of managing the download process yourself.*
+
+Before proceeding with the tutorial, you will have to make a few system configurations in order to be able to move on in a smooth manner.
+
+##### CA Certificates
+
+Since Haiku images can be fetched securely only via [HTTPS](https://en.wikipedia.org/wiki/HTTPS), you will need to download some [CA root certificates](https://en.wikipedia.org/wiki/Certificate_authority), as FreeBSD does not ship with them by default. In order to do that, you will need to install the `security/ca_root_nss` package, which is a part of the [FreeBSD ports](https://www.freebsd.org/ports/).
+
+```sh
+# feel free to skip this process if you have done it before.
+portsnap fetch
+portsnap extract
+
+cd /usr/ports/security/ca_root_nss
+make install clean
+```
+
+Now, follow through the process, and you will be able to fetch the image file securely.
+
+##### Networking & Kernel Preparations <a name="part_download"></a>
+
+Since `bhyve` consists of kernel modules that ship with FreeBSD by default, you do not need to install the core part of `bhyve`.
+
+However, you will need to load the modules manually in order to use `bhyve`.
+
+```sh
+kldload vmm
+kldload nmdm
+kldload if_bridge
+kldload if_tap
+```
+
+However, the kernel modules will only remain loaded until you reboot your machine. In order to make the changes permanent, edit `/boot/loader.conf` by adding:
```

Comment:
Maybe skip the "However" here, as you already have one in the previous sentence.